be/src/vec/exec/vaggregation_node.cpp

```diff
@@ -897,7 +963,40 @@ Status AggregationNode::_pre_agg_with_serialized_key(doris::vectorized::Block* i
     return Status::OK();
 }
 
+Status AggregationNode::_execute_with_serialized_key_limited(Block* block) {
+    SCOPED_TIMER(_build_timer);
```

Comment:
We should use templates to abstract the code of this part of the function to reduce the appearance of duplicate code
like
```
<bool limit>
_execute_with_serialized_key(Block* block) 
```

use limit to check whether to do `find` or `emplace`